test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -68,32 +71,81 @@ func DeletePod(mdb mdbv1.MongoDB, podNum int) func(*testing.T) {
 	}
 }
 
-// BasicConnectivity performs a check by initializing a mongo client
-// and inserting a document into the MongoDB resource
-func BasicConnectivity(mdb mdbv1.MongoDB) func(t *testing.T) {
+// BasicConnectivity returns a test function which performs
+// a basic MongoDB connectivity test
+func BasicConnectivity(mdb *mdbv1.MongoDB) func(t *testing.T) {
 	return func(t *testing.T) {
-		ctx, _ := context.WithTimeout(context.Background(), 10*time.Minute)
-		mongoClient, err := mongo.Connect(ctx, options.Client().ApplyURI(mdb.MongoURI()))
+		_, err := IsReachable(mdb)
 		if err != nil {
-			t.Fatal(err)
+			t.Fatal(fmt.Sprintf("Error connecting to MongoDB deployment: %+v", err))
 		}
+		t.Logf("successfully connected to MongoDB deployment")
+	}
+}
 
-		t.Logf("Created mongo client!")
-
-		var res *mongo.InsertOneResult
-		err = wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
-			collection := mongoClient.Database("testing").Collection("numbers")
-			res, err = collection.InsertOne(ctx, bson.M{"name": "pi", "value": 3.14159})
-			if err != nil {
-				t.Logf("error inserting document: %+v", err)
-				return false, err
-			}
-			return true, nil
-		})
+// IsReachable performs a check by initializing a mongo client
+// and inserting a document into the MongoDB resource
+func IsReachable(mdb *mdbv1.MongoDB) (bool, error) {
```

Comment:
This function has 2 return parameters that are dependant and I think one of them should be removed.

Under no circumstances the following can be observed, `true, err != nil` and that indicates me that we might get all we need by just returning `bool`.

So basically the truth value of `bool` is an alias (almost) of the value of error. And it is actually masking it on line #99, making an actual error be returned as a `false` value.

Can this be made into a 1-parameter returning function instead?

Reply:
yes you're right! We don't need to have the multi return here at all, nice catch 👍

Reply:
ah just seeing the one use case, the error provides the information about why it failed which is logged, in this case what do you think about just returning the error. Maybe with a different name. "Connect" or something similar which returns just an error

Reply:
I think it makes sense if you want to surface the error to the caller